### PR TITLE
LIMS-4793:Orders: Test Plan: Test unit pop up Approach: Make sure the test units that displayed below test plan, it belongs to this test plan

### PR DESCRIPTION
**Pull request by omnia353**
```
C:\Users\omnia\Documents\1limsNewMaster\1lims-automation>nosetests -vs --logging-level=WARNING ui_testing/testcases/basic_tests/test006_orders.py -m test103 --tc-file=config.ini
c:\users\omnia\lib\site-packages\nose\plugins\manager.py:394: RuntimeWarning: Unable to load plugin noseprogressive = noseprogressive:ProgressivePlugin: No module named 'fcntl'
  warn("Unable to load plugin %s: %s" % (ep, e),
2020-09-22 20:01:19.895 | INFO     | api_testing.apis.base_api:_get_authorized_session:46 - Get authorized api session.
2020-09-22 20:01:19.898 | INFO     | api_testing.apis.base_api:_get_authorized_session:47 - admin:admin
2020-09-22 20:01:20.661 | INFO     | api_testing.apis.base_api:_get_authorized_session:58 - session ID : IcuYiQyPuesYxr9RFtVzCk3bbUSNif8W9s2KtiIeiM8 .....

DevTools listening on ws://127.0.0.1:58915/devtools/browser/9cbfbd1a-c86c-47e8-ad59-c85cace9bcd8
Orders: Test Plan: Test unit pop up Approach: Make sure the test units that displayed below test plan, it belongs to this test plan ...
2020-09-22 20:02:24.135 | INFO     | ui_testing.testcases.base_test:setUp:27 - Test case : test103_check_test_unit_displayed_correct_under_test_plan
2020-09-22 20:02:28.917 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:603 - wait until page is loaded
2020-09-22 20:02:28.950 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-22 20:02:29.787 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-22 20:02:30.656 | INFO     | api_testing.apis.test_unit_api:set_name_configuration_name_only:437 - set test unit configuration
2020-09-22 20:02:31.227 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-22 20:02:31.245 | INFO     | api_testing.apis.orders_api:set_configuration:498 - set order configuration
2020-09-22 20:02:32.237 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-22 20:02:32.246 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/materialTypes
2020-09-22 20:02:32.814 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:32.822 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/articles
2020-09-22 20:02:33.262 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:33.270 | INFO     | api_testing.apis.article_api:get_article_with_material_type:184 - search for article with material type Raw Material
2020-09-22 20:02:33.292 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test103_check_test_unit_displayed_correct_under_test_plan:3471 - create 3 completed testplans
2020-09-22 20:02:33.304 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/materialTypes
2020-09-22 20:02:33.488 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:33.499 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits
2020-09-22 20:02:33.676 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:33.684 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits/get/304
2020-09-22 20:02:34.081 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:34.088 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testPlans
2020-09-22 20:02:34.485 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:34.493 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testPlans/get/10222
2020-09-22 20:02:34.630 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:34.649 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/materialTypes
2020-09-22 20:02:34.895 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:34.903 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits
2020-09-22 20:02:35.311 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:35.319 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits/get/305
2020-09-22 20:02:35.530 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:35.539 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testPlans
2020-09-22 20:02:35.834 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:35.839 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testPlans/get/41117
2020-09-22 20:02:36.038 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:36.047 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/materialTypes
2020-09-22 20:02:36.239 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:36.247 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits
2020-09-22 20:02:36.450 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:36.460 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits/get/306
2020-09-22 20:02:36.749 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:36.767 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testPlans
2020-09-22 20:02:37.000 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:37.008 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testPlans/get/75465
2020-09-22 20:02:37.253 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 20:02:37.261 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test103_check_test_unit_displayed_correct_under_test_plan:3477 - create order with testplans eabef3a33d,bc183478a6,d0d73b1536
2020-09-22 20:02:37.280 | INFO     | ui_testing.pages.order_page:create_new_order:140 -  Create new order.
2020-09-22 20:02:37.288 | INFO     | ui_testing.pages.orders_page:click_create_order_button:23 - Press create order button
2020-09-22 20:02:37.677 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-22 20:02:47.587 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-22 20:02:49.484 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-22 20:02:51.325 | INFO     | ui_testing.pages.order_page:set_new_order:19 - Set new order.
2020-09-22 20:02:58.956 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-22 20:03:06.527 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-22 20:03:14.123 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-22 20:03:15.814 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-22 20:03:25.217 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-22 20:03:37.503 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-22 20:04:03.372 | INFO     | ui_testing.pages.base_pages:save:55 - save the changes
2020-09-22 20:04:03.376 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-22 20:04:04.524 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-22 20:04:05.464 | INFO     | ui_testing.pages.order_page:create_new_order:178 -  Order created with no : 116-2020 
2020-09-22 20:04:05.958 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-09-22 20:04:08.899 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test103_check_test_unit_displayed_correct_under_test_plan:3482 - assert test unit : 222c01335a is displayed correctly under its
corresponding testplan :eabef3a33d 
2020-09-22 20:04:08.905 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test103_check_test_unit_displayed_correct_under_test_plan:3482 - assert test unit : b8c405b003 is displayed correctly under its
corresponding testplan :bc183478a6 
2020-09-22 20:04:08.910 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test103_check_test_unit_displayed_correct_under_test_plan:3482 - assert test unit : 81cdce7e63 is displayed correctly under its
corresponding testplan :d0d73b1536 
2020-09-22 20:04:08.912 | INFO     | ui_testing.testcases.base_test:tearDown:33 - go to dashboard page
2020-09-22 20:04:11.344 | INFO     | ui_testing.testcases.base_test:tearDown:35 - TearDown.     
ok

----------------------------------------------------------------------
Ran 1 test in 172.512s

OK

```

### CODE REVIEW

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 1]
```diff
@@ -3538,6 +3538,33 @@ def test100_create_multiple_suborders_with_testplans_testunits(self):
...
+        article_data = ArticleAPI().get_article_with_material_type(material_type='Raw Material')
+        formatted_article = {'id': article_data['id'], 'text': article_data['name']}
+        tp_list = []
+        tp_names = []
+        self.info('create 3 completed testplans')
+        for _ in range(3):
+            tp1 = TestPlanAPI().create_completed_testplan(material_type='Raw Material',
+                                                          formatted_article=formatted_article)
+            tp_names.append(tp1['testPlanEntity']['name'])
+            tp_list.append(tp1)
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test006_orders.py:3557]: you can use this function https://github.com/modeso-open-source/1lims-automation/blob/984d49541f1820a611a77166ab496bc13bf67df1/api_testing/apis/test_plan_api.py#L396
to create 3 test plans with Raw material and same article.

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 2]
```diff
@@ -3538,6 +3538,33 @@ def test100_create_multiple_suborders_with_testplans_testunits(self):
...
+        self.info('create order with testplans {},{},{}'.format(tp_names[0], tp_names[1], tp_names[2]))
+        self.order_page.create_new_order(material_type='Raw Material', test_units=[], test_plans=tp_names,
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test006_orders.py:3559]: then create order using API

**Review by ghada-37 — CHANGES_REQUESTED** [review]

File: ui_testing/testcases/basic_tests/test006_orders.py (hunk 2 above)
Comment (omnia353) [reply ui_testing/testcases/basic_tests/test006_orders.py:3559]: @ghada-37  if i used the above function -->
 def create_multiple_test_plan_with_same_article(self, no_of_testplans=2):

i will need to make changes in payload in order api

File: ui_testing/testcases/basic_tests/test006_orders.py (hunk 1 above)
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test006_orders.py]: I updated it now

File: ui_testing/testcases/basic_tests/test006_orders.py (hunk 2 above)
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test006_orders.py]: I made it now

**Comment by ghada-37** [comment]
@0xIslamTaha I refactored this PR using API, so I need your review on it please.

**Pull request closed without merging**